src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/OrdinaryMethodReferenceFinder.cs

```diff
@@ -65,38 +68,32 @@ private static ImmutableArray<ISymbol> GetOtherPartsOfPartial(IMethodSymbol symb
         // searches for these, then we should find usages of 'lock(goo)' or 'synclock(goo)'
         // since they implicitly call those methods.
 
-        var ordinaryDocuments = await FindDocumentsAsync(project, documents, cancellationToken, methodSymbol.Name).ConfigureAwait(false);
-        var forEachDocuments = IsForEachMethod(methodSymbol)
-            ? await FindDocumentsWithForEachStatementsAsync(project, documents, cancellationToken).ConfigureAwait(false)
-            : [];
+        await FindDocumentsAsync(project, documents, processResult, processResultData, cancellationToken, methodSymbol.Name).ConfigureAwait(false);
 
-        var deconstructDocuments = IsDeconstructMethod(methodSymbol)
-            ? await FindDocumentsWithDeconstructionAsync(project, documents, cancellationToken).ConfigureAwait(false)
-            : [];
+        if (IsForEachMethod(methodSymbol))
+            await FindDocumentsWithForEachStatementsAsync(project, documents, processResult, processResultData, cancellationToken).ConfigureAwait(false);
 
-        var awaitExpressionDocuments = IsGetAwaiterMethod(methodSymbol)
-            ? await FindDocumentsWithAwaitExpressionAsync(project, documents, cancellationToken).ConfigureAwait(false)
-            : [];
+        if (IsDeconstructMethod(methodSymbol))
+            await FindDocumentsWithDeconstructionAsync(project, documents, processResult, processResultData, cancellationToken).ConfigureAwait(false);
 
-        var documentsWithGlobalAttributes = await FindDocumentsWithGlobalSuppressMessageAttributeAsync(
-            project, documents, cancellationToken).ConfigureAwait(false);
+        if (IsGetAwaiterMethod(methodSymbol))
+            await FindDocumentsWithAwaitExpressionAsync(project, documents, processResult, processResultData, cancellationToken).ConfigureAwait(false);
 
-        var documentsWithCollectionInitializers = IsAddMethod(methodSymbol)
-            ? await FindDocumentsWithCollectionInitializersAsync(project, documents, cancellationToken).ConfigureAwait(false)
-            : [];
+        await FindDocumentsWithGlobalSuppressMessageAttributeAsync(
+            project, documents, processResult, processResultData, cancellationToken).ConfigureAwait(false);
 
-        return ordinaryDocuments.Concat(
```

Comment:
some oft hese concats may be able to  go away.

Reply:
Not clear on where you are talking about. The ones in this method did go away.

Reply:
i'm saying that the Concat extension methods can possibly go away.

Reply:
Doesn't look like there are a lot of callers of those, but it's under the compilers folder, so definitely not as part of this pr.

Reply:
WFM.